Libplanet/Net/Swarm.cs

```diff
@@ -1238,111 +1240,138 @@ bool render
+                            }
+
+                            count = 0;
+                            totalCount = recentStates.BlockStates.Count;
+                            _logger.Debug(
+                                "Starts to store block states received from {Peer}.",
+                                peer);
+                            foreach (var pair in recentStates.BlockStates)
+                            {
+                                cancellationToken.ThrowIfCancellationRequested();
+                                _store.SetBlockStates(pair.Key, pair.Value);
+                                progress?.Report(new BlockStateDownloadState()
                                 {
-                                    TotalStateReferenceCount = totalCount,
-                                    ReceivedStateReferenceCount = ++count,
+                                    TotalBlockStateCount = totalCount,
+                                    ReceivedBlockStateCount = ++count,
+                                    ReceivedBlockHash = pair.Key,
                                 });
                             }
                         }
-
-                        count = 0;
-                        totalCount = recentStates.BlockStates.Count;
-                        _logger.Debug("Starts to store block states received from {Peer}.", peer);
-                        foreach (var pair in recentStates.BlockStates)
+                        finally
                         {
-                            cancellationToken.ThrowIfCancellationRequested();
-                            _store.SetBlockStates(pair.Key, pair.Value);
-                            progress?.Report(new BlockStateDownloadState()
-                            {
-                                TotalBlockStateCount = totalCount,
-                                ReceivedBlockStateCount = ++count,
-                                ReceivedBlockHash = pair.Key,
-                            });
+                            rwlock.ExitWriteLock();
                         }
+
+                        _logger.Debug(
+                            "Stored state refs and block states. {Offset}",
+                            offset);
+
+                        offset = recentStates.Offset;
                     }
-                    finally
+                    else
                     {
-                        rwlock.ExitWriteLock();
+                        _logger.Debug(
+                            "A message received from {Peer} is not a RecentStates but {Reply}.",
+                            peer,
+                            reply
+                        );
+                        break;
                     }
+                }
 
-                    _logger.Debug(
-                        "Finished to store received state refs and block states.");
+                if (offset == -1)
+                {
+                    _logger.Warning("Received states from trusted peers.");
```

Comment:
This message seems normal case's, right?

Reply:
What do you mean by normal case?

Reply:
`"Received states from trusted peers."` doesn't seem a warning message. 😕

Reply:
You are right. I'll fix it :)